### PR TITLE
Fixes for eyewear and other rigid issues

### LINKED ISSUE #56384: Fit-over sunglasses cannot be worn over corrective glasses
### Describe the bug

Fit-over sunglasses are no longer wearable over corrective glasses due to them both being classified as rigid. Because the game strictly prevents you from wearing two rigid items on the same body part, they can no longer be worn together, even though fit-over sunglasses are explicitly designed to be worn over corrective glasses.

### Steps to reproduce

1. Spawn in some corrective sunglasses and fit-over sunglasses.
2. Put on one item or the other.
3. Observe that you can no longer wear both items at the same time; you'll get a notification that you can't wear two rigid items over your left eye.

### Expected behavior

Fit-over sunglasses are designed specifically to be worn over corrective lenses, hence the name. They should be wearable together.

### Screenshots

image
image

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19044.1586 (21H2)
- Game Version: https://github.com/CleverRaven/Cataclysm-DDA/commit/0cc5c0532ed9c6cade64d035cc83fced659604a8 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Blaze Industries [blazeindustries],
    No Rail Stations [No_Rail_Stations],
    Bionic Slots [cbm_slots],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth]
]

### Additional context

I'm fairly certain that this is a direct result of #55566.

### LINKED ISSUE #56391: Rigid Oopsies
### Is your feature request related to a problem? Please describe.

After the rigid armor changes some stuff may not work that should.

Please post anything that isn't working bellow and I will fix it, I will also tag relevant issues.

#56384

STUFF TO FIX:

- [x] Fit-over sunglasses 
- [x] Ballistic glasses 
- [x] welding goggles
- [x] safety glasses
- [x] riot & motorcycle helmet w/ visor down
- [x] eclipse glasses
- [x] PBA Mask
- [x] Rain Hood
- [x] Nomad Body Mesh
- [x] Second Skin
- [x] Medical Gloves
- [x] Rubber Gloves
- [x] Hazmat Suit
- [x] Emergency Blanket
- [x] Magic Rings
- [x] Masks
- [x] Jackets (rain, winter, windbreaker etc.)
- [x] Optical Cloak
- [ ] Army pants
- [ ] mil flight helmet 

### Solution you would like.

I'm gonna have to add some additional detain for stuff like wear over sunglasses and gas masks so they can layer but still conflict with each other.

### Describe alternatives you have considered.

Having future sight and knowing everything that will not work well.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by bombasticSlacks**
#### Summary
Bugfixes "Fixes for eyewear and other rigid issues"

#### Purpose of change
fixes #56391 fixes #56384

#### Describe the solution
Plan currently is to move some glasses around so eye layering makes sense in the first place, make some stuff soft that should be soft, and add an armor portion entry for stuff like eyewear, where you can wear something rigid on each layer separately since they don't touch.

So for stuff like eyes and head where armor and items will dangle without touching I've added an armor_portion entry for "rigid_layer_only" what this does is makes it so the item only considers rigidity on it's layer exclusively.

I've moved basic glasses to the inner layer and wear overs to the normal layer (same as gas masks and stuff). So you can do 
Glasses - Gas Mask Of Some Kind - Armor Visor which based on references is very possible. 

I've made alien liquid soft by default,

added flags to the rain hood, migo suit, and nomad suits that should be soft.

#### Additional context
there are some UI sticking points, I will address those in a separate PR. This is more of an immediate fix for issues.

### CODE REVIEW

**Comment by dseguin** [comment]
Hey Slacks, it looks like #55566 also triggered some issues with sort-of-armors like the shoulder strap (see #56389, #56382, #56399)

The fix is really simple, if you want to include it:

```diff
diff --git a/src/item.cpp b/src/item.cpp
index 52227f1231..e2e845840e 100644
--- a/src/item.cpp
+++ b/src/item.cpp
@@ -3593,16 +3593,18 @@ void item::armor_info( std::vector<iteminfo> &info, const iteminfo_query *parts,
     if( parts->test( iteminfo_parts::ARMOR_RIGIDITY ) && is_comfortable() ) {
         // if the item has no armor data it doesn't cover that part
         const islot_armor *armor = find_armor_data();
-        if( armor->comfortable ) {
-            std::string coverage = _( "<bold>This armor is comfortable</bold>" );
-            info.emplace_back( "ARMOR", coverage );
-        } else {
-            //only some parts are comfortable
-            std::string coverage = _( "<bold>Comfortable</bold>:" );
-            for( const armor_portion_data &entry : armor->sub_data ) {
-                if( entry.comfortable ) {
-                    for( const sub_bodypart_str_id &sbp : entry.sub_coverage ) {
-                        coverage += string_format( _( ", <info>%s</info>" ), sbp->name );
+        if( !!armor ) {
+            if( armor->comfortable ) {
+                std::string coverage = _( "<bold>This armor is comfortable</bold>" );
+                info.emplace_back( "ARMOR", coverage );
+            } else {
+                //only some parts are comfortable
+                std::string coverage = _( "<bold>Comfortable</bold>:" );
+                for( const armor_portion_data &entry : armor->sub_data ) {
+                    if( entry.comfortable ) {
+                        for( const sub_bodypart_str_id &sbp : entry.sub_coverage ) {
+                            coverage += string_format( _( ", <info>%s</info>" ), sbp->name );
+                        }
                     }
                 }
             }
```

File: src/item.h [hunk 1]
```diff
@@ -1559,6 +1559,9 @@ class item : public visitable
         template <typename T>
         bool is_bp_rigid( const T &bp ) const;
         template <typename T>
+        // check if rigid and that it only cares about the layer it is on
+        bool is_bp_rigid_selective( const T &bp ) const;
```
Comment (BrettDong) [inline src/item.h:1563]:
```suggestion
        // check if rigid and that it only cares about the layer it is on
        template <typename T>
        bool is_bp_rigid_selective( const T &bp ) const;
```

**Comment by Theawesomeboophis** [comment]
Cloaks can't be worn with backpacks, not sure if this is intentional.

**Comment by bombasticSlacks** [comment]
Which cloaks? The optical cloak?

**Comment by Theawesomeboophis** [comment]
> Which cloaks? The optical cloak?

Yes.

**Pull request merged**